File: python/helpers/cernoxThermom.py
import numpy as np

class cernoxThermom:
     def __init__(self, ZL, ZU, A):
          self.ZL = ZL
          self.ZU = ZU
          self.A = A

     def k(self, R):
          return(((np.log10(R) - self.ZL) - (self.ZU - np.log10(R))) / (self.ZU - self.ZL))

     def temp(self, R):
          return(sum(np.multiply(self.A, np.cos(np.arange(len(self.A)) * np.arccos(self.k(R))))))

File: matlabMeas/MattMeasurementFunctions/resThermo/resThermoHelpers/cernoxThermom.m
classdef cernoxThermom
    properties
        ZL;
        ZU;
        A;
    end
    methods
        function ct = cernoxThermom(ZL, ZU, A)
            ct.ZL = ZL;
            ct.ZU = ZU;
            ct.A = A;
        end

        function k = kval(obj, R)
            k = ((log10(R) - obj.ZL) - (obj.ZU - log10(R))) / (obj.ZU - obj.ZL);
        end

        function T = temp(obj, R)
            argint = cos((0:(length(obj.A) - 1)) * acos(kval(obj, R)));
            T = sum(obj.A .* argint);
        end
    end
end